### ISSUE TITLE
Gradients are NaN when using 512x512_aesthetic_0.json template file

### ISSUE THREAD

**Issue by Ericxgao** [deforum-art/deforum-stable-diffusion#53]
Have been trying to make my own losses, but seeing some odd results where the gradients are NaN whenever I set `gradient_wrt="x"`

Config:
```
{
    "W": 512,
    "H": 512,
    "sampler": "euler",
    "dynamic_threshold": null,
    "static_threshold": null,
    "mean_scale": 0,
    "var_scale": 0,
    "exposure_scale": 0,
    "exposure_target": 0,
    "colormatch_scale": 0,
    "ignore_sat_weight": null,
    "clip_scale": 0,
    "aesthetics_scale": 3,
    "init_mse_scale": 0,
    "blue_scale": 0,
    "gradient_wrt": "x",
    "gradient_add_to": "both",
    "decode_method": "autoencoder",
    "grad_threshold_type": "dynamic",
    "clamp_grad_threshold": 0.2,
    "grad_inject_timing": null,
    "cond_uncond_sync": true,
    "animation_prompts": {
        "0": "full body portrait of a woman with short wavy hair, round face, cottagecore!!, lake water, intricate, enlightenment, highly detailed, digital painting, artstation, concept art, smooth, sharp focus, illustration, art by artgerm and greg rutkowski and alphonse mucha"
    },
    "animation_mode": "2D",
    "max_frames": 50,
    "diffusion_cadence": "1"
}
```
Bad Gradients:
```
(DiffusionRunner pid=11539) tensor([[[[nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           ...,
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan]],
(DiffusionRunner pid=11539) 
(DiffusionRunner pid=11539)          [[nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           ...,
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan]],
(DiffusionRunner pid=11539) 
(DiffusionRunner pid=11539)          [[nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           ...,
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan]],
(DiffusionRunner pid=11539) 
(DiffusionRunner pid=11539)          [[nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           ...,
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan],
(DiffusionRunner pid=11539)           [nan, nan, nan,  ..., nan, nan, nan]]]], device='cuda:0')
(DiffusionRunner pid=11539) tensor(12.6680, device='cuda:0', grad_fn=<MulBackward0>)
(DiffusionRunner pid=11539) Loss: 12.66796875
```

Simple to repro - check the losses within the cond_fn:

```
    def make_cond_fn(self, loss_fn, scale):
        # Turns a loss function into a cond function that is applied to the decoded RGB sample
        # loss_fn (function): func(x, sigma, denoised) -> number
        # scale (number): how much this loss is applied to the image

        # Cond function with respect to x
        def cond_fn(x, sigma, denoised, **kwargs):
            with torch.enable_grad():
                denoised_sample = self.decode_fn(denoised).requires_grad_()
                loss = loss_fn(denoised_sample, sigma, **kwargs) * scale
                grad = -torch.autograd.grad(loss, x)[0]
                print(grad)
                print(loss)
            self.verbose_print('Loss:', loss.item())
            return grad
```
